### PR TITLE
Moving Brier and PIT metrics from RAIL to qp

### PR DESCRIPTION
**Pull request by drewoldag**
- WIP - Moved Brier metric from RAIL to qp.

### CODE REVIEW

File: src/qp/metrics/metrics.py [hunk 1]
```diff
@@ -1,70 +1,143 @@
 """This module implements some performance metrics for distribution parameterization"""
```
Comment (drewoldag) [inline src/qp/metrics/metrics.py:1]: This diff looks really intimidating, but that's almost entirely due to me alphabetizing the functions in this module. The new code that needs to be reviewed in is the following functions:
`calculate_anderson_ksamp`
`calculate_cramer_von_mises`
`calculate_kolmogorov_smirnov `
`calculate_outlier_rate`
`_check_ensembles_are_same_size` and 
`_check_ensemble_is_not_nested`

File: src/qp/pdf_gen.py [hunk 2]
```diff
@@ -321,7 +321,7 @@ def _argcheck(self, *args):
         return np.atleast_1d(cond)
 
     def freeze(self, *args, **kwds):
-        """Freeze the distribution for the given arguments.9999999
```
Comment (drewoldag) [inline src/qp/pdf_gen.py:324]: This one was just bugging me.

File: src/qp/metrics/metrics.py (hunk 1 above)
Comment (delucchi-cmu) [reply src/qp/metrics/metrics.py:1]: To avoid scary changes that aren't meaningful changes, I suggest first having a "prefactor" cleanup PR, where you do the alphabetizing and other semi-automated changes, so that more interesting changes are understandable (and cleanup PRs can be ignored in blame).

File: src/qp/metrics/array_metrics.py [hunk 3]
```diff
@@ -56,6 +64,45 @@ def quick_kld(p_eval, q_eval, dx=0.01):
...
+    p_eval: numpy.ndarray, float
+        the values of a probability distribution
+    grid: numpy.ndarray, float
+        the grid upon which p_eval was evaluated
+    dx: float
+        the difference between regular grid points
+    N: int
+        order of the moment to be calculated
+
+    Returns
```
Comment (OliviaLynn) [inline src/qp/metrics/array_metrics.py:98]: How important is consistency in comment styling (ie, having "Returns:" in some functions, and "Returns\n_______" in others)?

File: src/qp/metrics/pit.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.cramervonmises.html
+
+        Returns:
+            output [Objects]: A array of objects with attributes `statistic` and `pvalue`
+            For details see: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.cramervonmises.html
+        """
+        return stats.cramervonmises(self._pit_samps, 'uniform')
+
+    def evaluate_PIT_KS(self):
+        """_summary_
```
Comment (OliviaLynn) [inline src/qp/metrics/pit.py:98]: \_comment\_

**Review by OliviaLynn — APPROVED** [review]

File: src/qp/metrics/metrics.py (hunk 1 above)
Comment (drewoldag) [reply src/qp/metrics/metrics.py:1]: Thanks, that's a great idea! I'll go ahead and back out the alphabetization here, just to make this file more tractable.

File: src/qp/metrics/array_metrics.py (hunk 3 above)
Comment (drewoldag) [reply src/qp/metrics/array_metrics.py:98]: Thanks, that's a good question. I've been using an extension for VSCode called [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) that will create a template with blanks to fill in that theoretically will work well with [Sphinx](https://www.sphinx-doc.org/en/master/), a tool that will automatically generate nice looking python html docs (like [this one for qp](https://qp.readthedocs.io/en/latest/qp.html))

So, long story short - I'm not really sure how important it is. I'll take a look and make sure that the format that I'm using will play nicely.

File: src/qp/metrics/pit.py (hunk 4 above)
Comment (drewoldag) [reply src/qp/metrics/pit.py:98]: Oops, I clearly missed this one :) I'll fix this.

File: src/qp/metrics/metrics.py (hunk 1 above)
Comment (drewoldag) [reply src/qp/metrics/metrics.py:1]: I un-alphabetized the functions, so now the only changes are actual substantial code changes.

File: src/qp/metrics/pit.py (hunk 4 above)
Comment (drewoldag) [reply src/qp/metrics/pit.py]: I updated the docstring here.

File: src/qp/metrics/array_metrics.py (hunk 3 above)
Comment (drewoldag) [reply src/qp/metrics/array_metrics.py:98]: Yeah, you're right. These docstrings are not correctly converted into the readthedocs format. Or at least they don't look very good. I'll go through and update them with the correct format. Thanks for catching this!

Comment (eacharles) [reply src/qp/metrics/array_metrics.py:98]: Yes, readthedocs is expecting the numpy convention for formatting.

Comment (drewoldag) [reply src/qp/metrics/array_metrics.py:116]: I've just pushed a commit to address that formatting. Thanks all.

**Comment by drewoldag** [comment]
Just to link to the qp issue. This PR is part of a series, meant to address: https://github.com/LSSTDESC/qp/issues/25

File: src/qp/metrics/pit.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        eval_grid ([float], optional): A strictly increasing array-like sequence in the range [0,1]. Defaults to DEFAULT_QUANTS.
+    """
+
+    def __init__(self, qp_ens, ztrue, eval_grid=DEFAULT_QUANTS):
+        """We will create a quantile Ensemble to store the PIT distribution, but also store the
+        full set of PIT samples as ancillary data of the (single PDF) ensemble.
+
+        Args:
+            qp_ens Ensemble: A collection of N distribution objects
+            ztrue [float]: An array-like sequence of N float values representing the known true value for each distribution
```
Comment (aimalz) [inline src/qp/metrics/pit.py:26]: For the purposes of qp, we should probably call this `true_vals` or something like that because it won't always be redshift.

**Review by aimalz — APPROVED** [review]
Nice work! I left one (semantic) comment.
